core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/WorkflowTools.java

```diff
@@ -0,0 +1,334 @@
+      if (unactivateResolvedState && event.getResolvedState() != null) {
+        eligibleStates.remove(event.getResolvedState().getName());
+      }
+
+      // Retrieves action's consequences
+      Action action = model.getAction(event.getActionName());
+      Consequences consequences = action.getConsequences();
+
+      // Find first consequence according to comparisons
+      Iterator<Consequence> iConsequences = consequences.getConsequenceList().iterator();
+
+      boolean verified = false;
+      while (!verified && iConsequences.hasNext()) {
+        consequence = iConsequences.next();
+        if (consequence.getItem() != null) {
+          Field fieldToCompare = instance.getFolder().getField(consequence.getItem());
+          if (fieldToCompare != null && fieldToCompare.getStringValue() != null) {
+            verified = consequence.isVerified(fieldToCompare.getStringValue());
+          }
+        } else {
+          verified = true;
+        }
+      }
+
+      SilverLogger.getLogger(WorkflowTools.class)
+          .info("Process action {0}: item = {1}, operator = {2}, value = {3}",
+              event.getActionName(), consequence.getItem(), consequence.getOperator(),
+              consequence.getValue());
+
+      // if no consequence is verified, then last one will be used
+      if (consequence.getKill()) {
+        for (final String state : states) {
+          removeAffectations(instance, model.getState(state));
+        }
+
+        // process external actions
+        processTriggers(consequence, instance, event);
+
+        return true;
+      }
+      State[] targetStates = consequence.getTargetStates();
+      State[] unsetStates = consequence.getUnsetStates();
+
+      // for each target state, set eligibility to true
+      for (final State targetState : targetStates) {
+        if (targetState != null) {
+          String name = targetState.getName();
+          eligibleStates.put(name, name);
+        }
+      }
+
+      // for each unset state, set eligibility to false
+      for (final State unsetState : unsetStates) {
+        if (unsetState != null) {
+          String name = unsetState.getName();
+          eligibleStates.remove(name);
+        }
+      }
+
+      // notify user
+      List<QualifiedUsers> notifiedUsersList = consequence.getNotifiedUsers();
+      for (QualifiedUsers notifiedUsers : notifiedUsersList) {
+        Actor[] actors = instance.getActors(notifiedUsers, null);
+        Task[] tasks = taskManager.createTasks(actors, instance);
+        String message;
+
+        for (int i = 0; i < actors.length; i++) {
+          message = notifiedUsers.getMessage();
+          if (message == null || message.length() == 0) {
+            message = action.getDescription("", "");
+          }
+
+          // check if sender has been hardcoded in the model
+          String senderId = notifiedUsers.getSenderId();
+          User forcedUser = null;
+          if (senderId != null) {
+            try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1141)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)